### PR TITLE
P4info update staged

### PR DESCRIPTION
**Pull request by antoninbas**
This is the multi-stage version of https://github.com/p4lang/PI/pull/109 as suggested by @teverman. This implements stage 1) of the update.

### CODE REVIEW

**Comment by antoninbas** [comment]
Also Travis regressions are working for this pull request because we didn't remove any field from the protobuf messages, nor did we remove any protobuf message. However, it is worth noting that the protobuf numbering of some fields changed.

**Comment by teverman** [comment]
Otherwise, this looks good, subject to any lingering issues from 109.

**Comment by antoninbas** [comment]
@teverman I added the comment

File: proto/p4/config/p4info.proto [hunk 1]
```diff
@@ -69,37 +97,46 @@ message MatchField {
...
   Preamble preamble = 1;
   repeated MatchField match_fields = 2;
   // even when the table is indirect (see implementation_id) below, this field
   // includes all possible actions for the table
-  repeated int32 action_ids = 3;
+  repeated uint32 action_ids = 3;
   // 0 (default value) means that the table does not have a const default action
-  int32 const_default_action_id = 4;
+  uint32 const_default_action_id = 4;
+  // a table may have a const default action, whose action parameter values can
```
Comment (wmohsin) [inline proto/p4/config/p4info.proto]: nit: Reflow the comment (I see >80 char lines)
[Here and below]

File: proto/p4/config/p4info.proto [hunk 2]
```diff
@@ -69,37 +97,46 @@ message MatchField {
...
   // implementations may be added in the future
-  int32 implementation_id = 5;
+  uint32 implementation_id = 6;
   // ids of the direct resources (if any) attached to this table; for now this
   // includes only direct counters and direct meters, but other resources may be
   // added in the future
-  repeated int32 direct_resource_ids = 6;
-  int64 size = 7;  // number of entries in table
+  repeated uint32 direct_resource_ids = 7;
+  int64 size = 8;  // number of entries in table
```
Comment (wmohsin) [inline proto/p4/config/p4info.proto]: nit: /number/max number/ ?
(Also below)

File: proto/p4/config/p4info.proto [hunk 3]
```diff
@@ -42,25 +57,38 @@ message Preamble {
...
+  string path = 2;
+  // an alias for the P4 object, probably shorter than its path. The only
+  // constraint is for it to be unique with respect to other P4 objects of the
+  // same type. By default, the compiler uses the shortest suffix of the path
+  // that uniquely indentifies the object. For example if the P4 program
+  // contains two tables with paths s.c1.t and s.c2.t, the default names will
+  // respectively be c1.t and c2.t. The P4 programmer may also override the
+  // default name for any P4 object (TBD). When resolving a P4 object id, an
+  // application should be able to indiscriminately use the path or the name.
+  string name = 3;
```
Comment (wmohsin) [inline proto/p4/config/p4info.proto]: I suggest calling this "alias" to better capture the intent and semantics of this field.
(discussed briefly with antonin on chat as well)

**Comment by antoninbas** [comment]
@wmohsin I addressed your review comments

**Comment by antoninbas** [comment]
@wmohsin @teverman I also renamed `path` back to `name` now that we have `alias`. The idea was to avoid breaking existing code. IMO `name` works just as well as `path`, but let me know if you have objections.

**Comment by teverman** [comment]
I just patched in the change and built our code.  There were some minor issues with int32->uint32 changes, but with minimal changes it builds and passes unit tests.

**Pull request merged**